projects/plugins/boost/app/class-jetpack-boost.php

```diff
@@ -226,6 +226,22 @@ public static function activate() {
 		if ( $page_cache_status->get() && Boost_Cache_Settings::get_instance()->get_enabled() ) {
 			Page_Cache_Setup::run_setup();
 		}
+
+		$modules_setup = new Modules_Setup();
+
+		/*
+		 * Check what modules are already active (from a previous activation for example).
+		 * If there are active modules, we need to ensure each module-related event is triggered again.
+		 */
+		$active_modules = $modules_setup->get_ready_active_optimization_modules();
```

Comment:
`get_ready_active_optimization_modules` might not be the right list. The ready check will exclude all the modules that are currently not ready. For example, cloud CSS is not ready until it's generated, but it should be activated to generate the critical CSS in the first place and become ready.

So, in this case, we want to get all modules with the active state and run the routine on them.

Reply:
Thanks for the pointer! I updated the PR to try to address that.